streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -0,0 +1,557 @@
+                Materialized.as((WindowBytesStoreSupplier) supplier);
+
+            if (cache) {
+                materialized.withCachingEnabled();
+            } else {
+                materialized.withCachingDisabled();
+            }
+
+            if (log) {
+                materialized.withLoggingEnabled(Collections.emptyMap());
+            } else {
+                materialized.withCachingDisabled();
+            }
+
+            builder
+                .stream(
+                    INPUT_TOPIC_NAME,
+                    Consumed.with(Serdes.Integer(), Serdes.Integer())
+                )
+                .groupByKey()
+                .windowedBy(TimeWindows.ofSizeWithNoGrace(WINDOW_SIZE))
+                .aggregate(
+                    () -> 0,
+                    (key, value, aggregate) -> aggregate + value,
+                    materialized
+                );
+        } else if (supplier instanceof SessionBytesStoreSupplier) {
+            final Materialized<Integer, Integer, SessionStore<Bytes, byte[]>> materialized =
+                Materialized.as((SessionBytesStoreSupplier) supplier);
+
+            if (cache) {
+                materialized.withCachingEnabled();
+            } else {
+                materialized.withCachingDisabled();
+            }
+
+            if (log) {
+                materialized.withLoggingEnabled(Collections.emptyMap());
+            } else {
+                materialized.withCachingDisabled();
+            }
+
+            builder
+                .stream(
+                    INPUT_TOPIC_NAME,
+                    Consumed.with(Serdes.Integer(), Serdes.Integer())
+                )
+                .groupByKey()
+                .windowedBy(SessionWindows.ofInactivityGapWithNoGrace(WINDOW_SIZE))
+                .aggregate(
+                    () -> 0,
+                    (key, value, aggregate) -> aggregate + value,
+                    (aggKey, aggOne, aggTwo) -> aggOne + aggTwo,
+                    materialized
+                );
+        } else {
+            throw new AssertionError("Store supplier is an unrecognized type.");
+        }
+
+        // Don't need to wait for running, since tests can use iqv2 to wait until they
+        // get a valid response.
+        kafkaStreams =
+            IntegrationTestUtils.getStartedStreams(
+                streamsConfiguration(
+                    cache,
+                    log,
+                    supplier.getClass().getSimpleName()
+                ),
+                builder,
+                true
+            );
+    }
+
+    @After
+    public void afterTest() {
+        kafkaStreams.close();
+        kafkaStreams.cleanUp();
+    }
+
+    @AfterClass
+    public static void after() {
+        CLUSTER.stop();
+    }
+
+    @Test
+    public void verifyStore() {
+        shouldRejectUnknownQuery();
+        shouldHandlePingQuery();
+        shouldCollectExecutionInfo();
+        shouldCollectExecutionInfoUnderFailure();
+    }
```

Comment:
This was a bit of a bummer for me. I was sad to learn that JUnit has no concept of Before/After Instance, so there's no way to set up the kafkaStreams application just once per store config and then run all the tests; it would have repeated the setup on each test in this class.

This is super valuable, since creating/destroying Streams can take up to a second, while running these checks just takes a few milliseconds. So we'd be spending most of our time in setup/teardown, and considering the number of parameters, it'd be a lot of time.

So, the workaround is to have just one `Test`, which calls all the checks we care about. It's slightly worse for debugging later, but the stacktrace will still tell us which check failed.